CMakeLists.txt

```diff
@@ -444,6 +444,11 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         list(APPEND warning_flags -Wno-deprecated-literal-operator)
     endif()
 
+    if(CMAKE_CXX_COMPILER_ID STREQUAL AppleClang AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 17)
```

Comment:
If possible can we follow the existing syntax of

`if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND APPLE`

Your change to test for `AppleClang` does look nicer but is different to the existing checks and it would be good to be consistent